src/core/thread/network_data_service.hpp

```diff
@@ -159,41 +159,46 @@ class Manager : public InstanceLocator, private NonCopyable
      *
      * @param[in] aAddress    The unicast address.
      * @param[in] aPort       The port number.
+     * @param[in] aVersion    The version.
      *
      * @retval kErrorNone     Successfully added the Service entry.
      * @retval kErrorNoBufs   Insufficient space to add the Service entry.
      */
-    Error AddDnsSrpUnicastServiceWithAddrInServiceData(const Ip6::Address &aAddress, uint16_t aPort)
+    Error AddDnsSrpUnicastServiceWithAddrInServiceData(const Ip6::Address &aAddress, uint16_t aPort, uint8_t aVersion)
     {
-        return AddService(DnsSrpUnicast::ServiceData(aAddress, aPort));
+        return AddService(DnsSrpUnicast::ServiceData(aAddress, aPort, aVersion));
     }
 
     /**
      * Removes a DNS/SRP Unicast Service entry with address in Service Data from the local Thread Network Data.
      *
      * @param[in] aAddress    The unicast address.
      * @param[in] aPort       The port number.
+     * @param[in] aVersion    The version.
```

Comment:
This will be helpful for testing and can have additional use cases.

I expect the version to be fixed, but it may not always be known (e.g., defined as a constant) within the OT stack itself. For example, some projects may choose to implement their own SRP server and DNS resolver functions and not use the OT stack implementations. We provide these APIs to help publish and unpublish DNS/SRP entries in Network Data.

Reply:
Thanks - I see, so it could be that the platform adds a new feature for its DNS services, which warrants a different version byte to be published - in that case it needs to be dynamically set on OT.